garrysmod/lua/vgui/dscrollpanel.lua

```diff
@@ -27,119 +30,106 @@ function PANEL:Init()
 	self:SetPaintBackgroundEnabled( false )
 	self:SetPaintBorderEnabled( false )
 	self:SetPaintBackground( false )
-
 end
 
 function PANEL:AddItem( pnl )
-
 	pnl:SetParent( self:GetCanvas() )
-
 end
 
 function PANEL:OnChildAdded( child )
-
 	self:AddItem( child )
-
 end
 
 function PANEL:SizeToContents()
-
-	self:SetSize( self.pnlCanvas:GetSize() )
-
+	local canvas = self:GetCanvas()
+	if IsValid( canvas ) then

```

Comment:
Could be changed to an early return

Reply:
I will do it, chance that it will not be there is extremely small...